File: app/src/main/java/com/egd/userinterface/controllers/models/IMotorController.java
package com.egd.userinterface.controllers.models;

/**
 * Abstraction for the controller responsible for managing the motor.
 *
 * @author Petar Krastev
 * @since 12.11.2017
 */
public interface IMotorController extends IController {

    /**
     * Starts the motor.
     */
    void start();

    /**
     * Stops the motor.
     */
    void stop();
}

File: app/src/main/java/com/egd/userinterface/controllers/models/IMotorController.kt
package com.egd.userinterface.controllers.models

/**
 * Abstraction for the controller responsible for managing the motor.
 *
 * @author Petar Krastev
 * @since 12.11.2017
 */
interface IMotorController : IController {

    /**
     * Starts the motor.
     */
    fun start()

    /**
     * Stops the motor.
     */
    fun stop()
}
